### DISCUSSION TITLE
How does this plugin integrate into neovim ?

### DISCUSSION THREAD

**Discussion by adam-devel** [nvim-treesitter/nvim-treesitter#7445] [Q&A]
I assume the highlighting in neovim is left to plugins that may use treesitter, regexes or whatever the author sees fit. I assume it's just code that gets executed.

Neovim already highlights a lot of languages without this plugin. (I assume) it already has built-in plugins for syntax highlighting for these languages.

The questions I have are the following:
- How doesn't this plugin clash with existing syntax highlighting plugins ?
- Maybe when this plugin is enabled it does some sort of unloading of the other syntax highlighting plugins or it overwrites their work ? there's a lot of waste in doing an initial syntax highlighting just for it to be discarded by the later enabled treesitter
- assuming it's the case there's a waste of performance in the initial syntax highlighting, there should be an option to disable builtin syntax highlighting plugins so tree sitter can take over

**Comment by clason** [comment]
This plugin only handles parser installation and provides queries for parsers not distributed with Neovim itself. Everything else is handled by Neovim itself (and I can assure you that the interaction with legacy syntax highlighting is carefully considered; by default it's disabled if you start treesitter highlighting).

**Reply by clason** [reply]
(You make a lot of unwarranted assumptions; I'd suggest reading the Neovim documentation about syntax and treesitter first.)

**Reply by adam-devel** [reply]
# according  to `:h treesitter`

Neovim assosiciates the first match found for a language with the parser at `parser/{lang}.*`. 

Treesitter queries are read from `queries/**.scm`. they are used for highlighting and other purposes.

Highlighting is triggered with`vim.treesitter.start()` using `queries/{lang}/highlighs.scm`.

Nvim-treesitter already has a `parser/` and `queries/` subdirectory, so it's parsers and queries are prioritized according to the runtimepath. Good !

# according to `:h syntax`

neovim sources `$VIMRUNTIME/syntax/syntax.vim`, which —in summary— sets up a chain of autocommands for recognizing the filetype, and finally triggering `runtime! syntax/<name>.vim`, loading the first syntax file found per filetype. then triggering the FileType and Syntax autocommads for plugins

It seems the only way to avoid triggering the bundled syntax highlighting plugins is by having your `syntax/*` be loaded earlier. the runtimepath already prioritizes user configs. Good !

but treesitter has no `syntax/` subdirectory, how does it avoid loading builtin syntax plugins ?

**Reply by clason** [reply]
Apples and oranges. These are two completely different subsystems.

**Reply by adam-devel** [reply]
you told me to read that documentation

> I'd suggest reading the Neovim documentation about syntax and treesitter

I read both and I summarized them in the the comment above. mentionning the relevant questions i still had.

I realize they are two different systems. they both can be used to provide syntax highlighting however, which is what's relevant in my original question. the question is: how does treesitter not clash with the built-in syntax highlighting plugins that do not rely on treesitter parsers and highlight queries ?
Maybe neovim doesn't load syntax files when a treesitter parser is present? if so that's not mentioned in the docs. or Maybe it's a neovim policy to only bundle treesitter based syntax highlighting ?
